File: java-jpa/src/main/java/edu/pkch/jpa/Post.java
package edu.pkch.jpa;

import javax.persistence.*;

@Entity
@Table(name = "post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 100)
    private String title;

    @Lob
    @Column
    private String content;

    @ManyToOne
    private Member member;

    protected Post() {}

    public Post(String title, String content, Member member) {
        this.title = title;
        this.content = content;
        this.member = member;
    }
}

File: kotlin-jpa/src/main/kotlin/edu/pkch/jpa/Post.kt
package edu.pkch.jpa

import javax.persistence.*

@Entity
@Table(name = "post")
class Post(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long,

    @Column
    var title: String,

    @Lob @Column
    var contents: String,

    @ManyToOne
    var member: Member
)